truss/contexts/image_builder/cache_warmer.py

```diff
@@ -48,106 +85,151 @@ def split_path(path, prefix="gs://"):
                 config=Config(signature_version="s3v4"),
             )
-            bucket_name, _ = split_path(bucket_name, prefix="s3://")
-
-            dst_file = Path(f"{cache_dir}/{file_name}")
-            if not dst_file.parent.exists():
-                dst_file.parent.mkdir(parents=True)
-
-            try:
-                url = client.generate_presigned_url(
-                    "get_object",
-                    Params={"Bucket": bucket_name, "Key": file_name},
-                    ExpiresIn=3600,
-                )
-            except Exception:
-                raise RuntimeError(f"File not found on S3 bucket: {file_name}")
-
-            try:
-                proc = _download_from_url_using_b10cp(_b10cp_path(), url, dst_file)
-                proc.wait()
-            except Exception as e:
-                raise RuntimeError(f"Failure downloading file from S3: {e}")
-    else:
-        secret_path = Path("/etc/secrets/hf-access-token")
-        secret = secret_path.read_text().strip() if secret_path.exists() else None
+        else:
+            client = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+
+        # Cache file
+        cache_dir = Path(f"/app/model_cache/{bucket_name}")
+        cache_dir.mkdir(parents=True, exist_ok=True)
+
+        dst_file = cache_dir / self.file_name
+        if not dst_file.parent.exists():
+            dst_file.parent.mkdir(parents=True)
+
         try:
-            hf_hub_download(
-                repo_name,
-                file_name,
-                revision=revision_name,
-                token=secret,
+            url = client.generate_presigned_url(
+                "get_object",
+                Params={"Bucket": bucket_name, "Key": file_name},
+                ExpiresIn=3600,
             )
-        except FileNotFoundError:
+        except NoCredentialsError as nce:
             raise RuntimeError(
-                "Hugging Face repository not found (and no valid secret found for possibly private repository)."
+                f"No AWS credentials found\nOriginal exception: {str(nce)}"
+            )
+        except ClientError as ce:
+            raise RuntimeError(
+                f"Client error when accessing the S3 bucket (check your credentials): {str(ce)}"
             )
+        except Exception as exc:
+            raise RuntimeError(
+                f"File not found on S3 bucket: {file_name}\nOriginal exception: {str(exc)}"
+            )
+
+        download_file_using_b10cp(url, dst_file, self.file_name)
+
+
+def download_file_using_b10cp(url, dst_file, file_name):
+    try:
+        proc = _download_from_url_using_b10cp(_b10cp_path(), url, dst_file)
+        proc.wait()
+
+    except FileNotFoundError as file_error:
+        raise RuntimeError(f"Failure due to file ({file_name}) not found: {file_error}")
+
+    except TimeoutError as timeout_error:
+        raise RuntimeError(f"Failure due to timeout: {timeout_error}")
```

Comment:
For TimeoutError, OSError, and ValueError, why not just throw that exception (Instead of catching and reraising a RuntimeError)? I don't think the RuntimeError adds anything here